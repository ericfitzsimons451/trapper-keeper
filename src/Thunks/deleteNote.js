import { deleteCard, isLoading, hasError } from "../actions";

export const deleteNote = (id) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(`http://localhost:3000/api/v1/notes/${id}`, {
        method: "DELETE"
      });
      if (response.status !== 204) {
        throw new Error(response.statusText);
      }
      dispatch(isLoading(false));
      dispatch(deleteCard(id));
    } catch (error) {
      console.log("hello world")
      dispatch(hasError(error.message));
    }
  };
};
