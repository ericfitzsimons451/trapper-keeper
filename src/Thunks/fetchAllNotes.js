import { getAllNotes, hasError, isLoading } from '../actions'

export const fetchAllNotes = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw new Error(response.statusText)
      }
      const notes = await response.json()
      dispatch(isLoading(false))
      dispatch(getAllNotes(notes))
    } catch (error) {
      dispatch(hasError(error.message))
    }
  }
}