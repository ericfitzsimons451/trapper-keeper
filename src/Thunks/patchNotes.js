import { hasError } from '../actions'

export const patchNotes = (notes) => {
	const option = {
		method: "PATCH",
		headers: {
			'Content-Type': 'application/json'
		},
		body: JSON.stringify(notes)
	}

	return async (dispatch) => {
		try {
			const response = await fetch("http://localhost:3000/api/v1/notes", option)
			if (!response.ok) {
				throw new Error(response.statusText)
			}
		} catch (error) {
			dispatch(hasError(error.message))
		}
	}
}