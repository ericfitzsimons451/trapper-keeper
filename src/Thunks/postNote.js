import { addNote, isLoading, hasError } from '../actions'

export const postNote = (note) => {
    return async (dispatch) => {
        try {
            dispatch(isLoading(true))
            const response = await fetch('http://localhost:3000/api/v1/notes')
            if (!response.ok) {
                throw new Error(response.statusText)
            }
            const note = await response.json()
            dispatch(isLoading(false))
            dispatch(addNote(note))
        } catch (error) {
            dispatch(hasError(error.message))
        }
    }
}