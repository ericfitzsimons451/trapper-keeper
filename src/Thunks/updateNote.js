import { updateNote, isLoading, hasError } from '../actions'

export const editNote = (noteToUpdate) => {
  console.log('Update Note Thunk Fires')
  const option = {
    method: 'PUT',
    body: JSON.stringify(noteToUpdate),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(`http://localhost:3000/api/v1/notes/${noteToUpdate.id}`, option)
      if (!response.ok) {
        throw new Error(response.statusText)
      }
      const note = await response.json()
      dispatch(isLoading(false))
      console.log(noteToUpdate)
      dispatch(updateNote(noteToUpdate))
    } catch (error) {
      dispatch(hasError(error.message))
    }
  }
}