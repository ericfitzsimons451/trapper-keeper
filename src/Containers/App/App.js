import React, { Component } from 'react';
import NoteForm from '../NoteForm/NoteForm'
import { NavLink, Route } from 'react-router-dom'
import './App.scss';
import { connect } from 'react-redux'
import { fetchAllNotes } from '../../Thunks/fetchAllNotes'
import { Header } from '../../Components/Header/Header'
import  NotesContainer  from '../NotesContainer/NotesContainer';

class App extends Component {
  async componentDidMount() {
    const url = 'http://localhost:3000/api/v1/notes'
    this.props.fetchAllNotes(url)
  }

  render() {
    return (
      <div className="App">
        <Header/>
        <NavLink to='/notes'>Create New Note</NavLink>
        <Route exact path='/notes' component={NoteForm} />
        <NotesContainer/>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  notes: state.notes,
  errorMsg: state.errorMsg,
  isLoading: state.isLoading
})

export const mapDispatchtoProps = dispatch => ({
  fetchAllNotes: (url) => dispatch(fetchAllNotes(url)),
})

export default connect(mapStateToProps, mapDispatchtoProps)(App);
