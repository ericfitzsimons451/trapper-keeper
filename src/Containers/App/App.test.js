import React from 'react';
import { shallow } from "enzyme";
import App, { mapStateToProps, mapDispatchtoProps } from './App'
import { fetchAllNotes } from '../../Thunks/fetchAllNotes';

jest.mock('../../Thunks/fetchAllNotes');

describe("App", () => {
  let wrapper;
  const mockFunc = jest.fn();
  beforeEach(() => { wrapper = shallow(<App fetchAllNotes={mockFunc}/>)
  
  })

  it.skip("should match the snapshot with all data passed in correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });


  it.skip("should invoke fetchAllNotes when componentDidMount is fired", () => {
    expect(fetchAllNotes).toHaveBeenCalled();
  })

  it("should map state to props", () => {
    const mockStore = {
      notes: [1,2,3],
      errorMsg: 'Error',
      isLoading: false
    }
    const expected = {
      notes: [1,2,3],
      errorMsg: 'Error',
      isLoading: false      
    }
    const mappedProps = mapStateToProps(mockStore)
    expect(mappedProps).toEqual(expected)
  });

  it.skip("should map dispatch to props", () => {
    const mockDispatch = jest.fn()
    const mockUrl = "www.reddit.com"
    const actionToDispatch = fetchAllNotes(mockUrl)
    const mappedProps = mapDispatchtoProps(mockDispatch)
    mappedProps.fetchAllNotes(mockUrl)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  });

})