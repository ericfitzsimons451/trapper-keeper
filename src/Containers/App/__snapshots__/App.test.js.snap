// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<div
  className="App"
>
  <Header />
  <Route
    component={[Function]}
    exact={true}
    path="/new-note"
  />
  <Route
    exact={true}
    path="/notes/:id"
    render={[Function]}
  />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
</div>
`;
